pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -248,13 +248,16 @@ def state_key(self) -> str:
             save_on_train_epoch_end=self._save_on_train_epoch_end,
         )
 
-    def on_pretrain_routine_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        """When pretrain routine starts we resolve the ckpt dir on the fly."""
+    def on_before_accelerator_backend_setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
```

Comment:
how about running this in the `setup` hook?

Reply:
yes agreed better than the accelerator_setup hook